### ISSUE TITLE
Surface `Warning` header

### ISSUE THREAD

**Issue by cben** [abonas/kubeclient#484]
Since 1.19, apiserver may set `Warning:` header on responses. https://kubernetes.io/blog/2020/09/03/warnings/

How may we expose it to callers?  
NOT an exception, this is for stuff like deprecated APIs that however _should still work_.
The common use case is probably to log them, but should be customizable.
So, add a param to `Client.new()` specifying a callback / logger object?

**Comment by astencel-sumo** [comment]
Yes I agree, being able to log the warning to a user-provided logger instance sounds like a good idea.
